### ISSUE TITLE
SimilarStepId is too strict

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#166]
### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - id: params
           run: 'true'
         - id: pages
           run: 'true'
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

SimilarStepId should allow it

### Actual behavior

SimilarStepId finds these two steps

### Additional Context

https://github.com/TWiStErRob/twisterrob.github.io/security/code-scanning/37
